src/main/scala/mesosphere/marathon/state/Volume.scala

```diff
@@ -96,11 +105,61 @@ object DockerVolume {
   }
 }
 
-case class PersistentVolumeInfo(size: Long)
+/**
+  * PersistentVolumeInfo captures the specification for a volume that survives task restarts.
+  *
+  * `name` is the *unique name* of the storage volume. names should be treated as case insensitive labels
+  * derived from an alpha-numeric character range [a-z0-9]. while there is no prescribed length limit for
+  * volume names it has been observed that some storage provider implementations may refuse names greater
+  * than 31 characters. YMMV. Although `name` is optional, some storage providers may require it.
+  *
+  * `name` uniqueness:
+  *  <li> A volume name MUST be unique within the scope of a volume provider.
+  *  <li> A fully-qualified volume name is expected to be unique across the cluster and may formed, for example,
+  *       by concatenating the volume provider name with the volume name. E.g “dvdi.volume123”
+  *
+  * `providerName` is optional; if specified it indicates which storage provider will implement volume
+  * lifecycle management operations for the persistent volume. if unspecified, “agent” is assumed.
+  * the provider names “dcos”, “agent”, and are currently reserved. The contents of providerName values
+  * are restricted to the alpha-numeric character range [a-z0-9].
+  *
+  * `options` contains provider-specific volume options. some items may be required in order for a volume
+  * driver to function properly. Consider a storage provider named “dvdi”. all options specific to that
+  * provider MUST be namespaced with a “dvdi/” prefix.
+  *
+  * DCOS-specific options will be prefixed with “dcos/”. an example of a DCOS option might be
+  * “dcos/label”, a user-assigned, human-friendly label that appears in a UI.
+  *
+  * future: will contain a mix of DCOS, Marathon, and provider-specific options.
+  *
+  * @param size absolute size of the volume (MB)
+  * @param name identifies the volume within the context of the storage provider.
+  * @param providerName identifies the storage provider responsible for volume lifecycle operations.
+  * @param options contains storage provider-specific configuration configuration
+  */
+case class PersistentVolumeInfo(
+  size: Option[Long] = None,
+  name: Option[String] = None,
+  providerName: Option[String] = None,
```

Comment:
for example, I removed this just prior to commit because I wasn't using it:
```scala
protected object Volumes { 
  val registry = Map[String, Volumes]()
  
  def register(vp: Volumes) = {
    registry(vp.name) = vp
  }

  def apply(name: String): Volumes = registry(name)

  // all volume providers should be registered here, statically
  register(AgentVolumes)
}
```